File: ios/Spandan/Sources/UI/OverlayView.swift
import UIKit

/// Draws the live face-detection box and the smaller ROI sub-box on top of the
/// camera preview. Coordinates are expected to already be in this view's own
/// point space (see CoordinateMapper.viewRect) -- this class only draws, it
/// doesn't do any coordinate math itself. Ported from
/// android/app/.../ui/OverlayView.kt.
final class OverlayView: UIView {

    private var faceRect: CGRect?
    private var roiRect: CGRect?

    private let faceColor = UIColor(red: 0x4C / 255, green: 0xAF / 255, blue: 0x50 / 255, alpha: 1) // green
    private let roiColor = UIColor(red: 0xFF / 255, green: 0xEB / 255, blue: 0x3B / 255, alpha: 1) // yellow

    override init(frame: CGRect) {
        super.init(frame: frame)
        commonInit()
    }

    required init?(coder: NSCoder) {
        super.init(coder: coder)
        commonInit()
    }

    private func commonInit() {
        backgroundColor = .clear
        isOpaque = false
        isUserInteractionEnabled = false
    }

    /// Pass nil for both to clear the overlay (e.g. when no face is found).
    func update(face: CGRect?, roi: CGRect?) {
        faceRect = face
        roiRect = roi
        setNeedsDisplay()
    }

    override func draw(_ rect: CGRect) {
        guard let ctx = UIGraphicsGetCurrentContext() else { return }

        if let face = faceRect {
            ctx.setStrokeColor(faceColor.cgColor)
            ctx.setLineWidth(2.5)
            ctx.stroke(face)
        }
        if let roi = roiRect {
            ctx.setStrokeColor(roiColor.cgColor)
            ctx.setLineWidth(2.0)
            ctx.stroke(roi)
        }
    }
}

File: android_segment30_mediapipe/app/src/main/java/com/spandan/app/ui/OverlayView.kt
package com.spandan.app.ui

import android.content.Context
import android.graphics.Canvas
import android.graphics.Color
import android.graphics.Paint
import android.graphics.RectF
import android.util.AttributeSet
import android.view.View

/**
 * Draws the live face-detection box and the smaller ROI sub-box on top of
 * the camera preview. Coordinates are expected to already be in this view's
 * own pixel space (see CoordinateMapper.rotatedRectToViewRect) -- this class
 * only draws, it doesn't do any coordinate math itself.
 */
class OverlayView @JvmOverloads constructor(
    context: Context,
    attrs: AttributeSet? = null
) : View(context, attrs) {

    private var faceRect: RectF? = null
    private var roiRect: RectF? = null

    private val facePaint = Paint().apply {
        color = Color.parseColor("#4CAF50") // green
        style = Paint.Style.STROKE
        strokeWidth = 6f
        isAntiAlias = true
    }

    private val roiPaint = Paint().apply {
        color = Color.parseColor("#FFEB3B") // yellow
        style = Paint.Style.STROKE
        strokeWidth = 5f
        isAntiAlias = true
    }

    /** Pass null for both to clear the overlay (e.g. when no face is found). */
    fun update(face: RectF?, roi: RectF?) {
        faceRect = face
        roiRect = roi
        postInvalidateOnAnimation()
    }

    override fun onDraw(canvas: Canvas) {
        super.onDraw(canvas)
        faceRect?.let { canvas.drawRect(it, facePaint) }
        roiRect?.let { canvas.drawRect(it, roiPaint) }
    }
}
